components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissions.kt

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.sitepermissions
+
+/**
+ * A site permissions and its state.
+ */
+data class SitePermissions(
+    val origin: String,
+    val location: Boolean = false,
+    val notification: Boolean = false,
+    val microphone: Boolean = false,
+    val cameraBack: Boolean = false,
+    val cameraFront: Boolean = false,
+    val bluetooth: Boolean = false,
+    val localStorage: Boolean = false,
+    val savedAt: Long
+)
```

Comment:
I don't think a `Boolean` is enough here; since we need to model "Allowed", "Blocked", "No decision". For example with `microphone = false` it's unclear whether the user has decided to block this or whether the site has never requested this.

Reply:
I didn't know that we will store **"No decision"** as a state.

I thought that we only will store the site permissions when they get **granted** or **denied**, and we will assume that if the `origin` is not in the db is because it hasn't be asked before (**"No decision"**). As this state only happens one time. Am I wrong?

Is this approach enough? Or should we still need to store  **"No decision"** as a state.
 Could you help me to clarify the use case for "**No decision"** and when we save it to the db?

This is the flow that I initially thought:

### Case 1:
A user enters for first time to a site that requires a notification permission:

We check if the `origin` is in the db using `store.findSitePermissionsBy(origin: String)`,
as it's the first time for the user, the `origin` is not in the db. Then we show the prompt.

 1. If the user hit the **"Allow"** button, then we save the origin using `store.save(sitePermissions: SitePermissions)`  with the `sitePermissions.notification` marked as  **"Allowed"** .

`Otherwise`

2.  If the user hit the **"Don't Allow"** button, then we save the origin using `store.save(sitePermissions: SitePermissions)`  with the `sitePermissions.notification` marked as  **"Blocked"** .


### Case 2:
The same user enters to the same site, but this time the `origin` is in the db with the `sitePermissions.notification` marked as  **"Allowed"**. 

1. We check if the `origin` is in the db using `store.findSitePermissionsBy(origin: String)`,
and **it's there** with the `sitePermissions.notification` marked as  **"Allowed"**. Then  we  don't show the prompt and grant the permission.

### Case 3:
The same user enters to the same site the `origin` is in the db with the `sitePermissions.notification` marked as  **"Blocked"**. 

1. We check if the `origin` is in the db using `store.findSitePermissionsBy(origin: String)`,
and **it's there** with the `sitePermissions.notification` marked as  **"Blocked"**. Then  we do show the prompt and let the user chooses.

Reply:
@jonalmeida helped me to clarify why  **"No decision"** is need it :)